baseline.gradle

```diff
@@ -45,7 +45,10 @@ subprojects {
   apply plugin: 'com.palantir.baseline-reproducibility'
   apply plugin: 'com.palantir.baseline-exact-dependencies'
   apply plugin: 'com.palantir.baseline-release-compatibility'
-  apply plugin: 'com.diffplug.spotless'
+  // We need to update Google Java Format to 1.17.0+ to run spotless on JDK 8, but that requires dropping support for JDK 8.
+  if (JavaVersion.current() != JavaVersion.VERSION_21) {
+    apply plugin: 'com.diffplug.spotless'
```

Comment:
I don't think we can merge the PR without having spotless enabled unfortunately

Reply:
I wish this was avoidable. I do not know of a way to run same formatting rules under Java 8 and Java 21 and that's we support today. 
I do believe that being able to test and actually testing with 21 is incremental improvement over current state. I see value in adding even partial support for 21. In particular, this PR identifies a problem with scala 2.12.

Reply:
Can you bring this topic up on the Dev mailing list to see what others think about adding JDK21 support without being able to run spotless?

Reply:
@jbonofre we can't upgrade the GJF as long as we support JDK8 because upgrading the GJF version will produce different results depending on the underlying JDK it is executed with and so CI actions (and locally) will fail.

Reply:
That's my point: I would postpone the JDK21 update right now, and jump directly to JDK21 without supporting JDK8 anymore.
I think it will be very difficult to support both JDK21 and JDK8 from spotless/GJF standpoint. So, I would directly jump to new version when possible.

Reply:
Dropping JDK8 support means dropping Hive support (which I'm totally fine with) but we need to properly communicate which Iceberg version will drop Hive support. My understanding was that we were planning to drop Hive support with Iceberg 2.0

Reply:
Yes exactly. It's what I proposed while ago in the Iceberg 2.0 discussion thread. So, I propose:
- on 1.x (including 1.6.0 release currently in preparation), we keep JDK8
- for 2.x, we merge this PR and jump to JDK21

I can reply on the Iceberg 2.0 thread if you want or Piotr can start a new thread about JDK21 on Iceberg 2.0.
I would love to have feedback on the Iceberg 2.0 thread 😄

Reply:
> Dropping JDK8 support means dropping Hive support (which I'm totally fine with) but we need to properly communicate which Iceberg version will drop Hive support. My understanding was that we were planning to drop Hive support with Iceberg 2.0

Sir. Maybe the iceberg-hive module should be maintained by the hive community in the hive project. Because hive 3.X will elo sooner or later.

Reply:
@BsoBird newer versions of Hive are maintained by the Hive community but the Iceberg community still maintains an older version of Hive